### PR TITLE
Add options, parse out nil options, and pass along stream_to options from new method

### PR DESCRIPTION
**Pull request by AtibaBryan**
That pretty much sums up this pr. My org is working on an elixir app that
requires an http streaming client for an api that only works over tls.
I figure bringing our changes into mainline would be helpful for everybody.

Let me know if there's anything you'd like me to change.

### CODE REVIEW

File: lib/eventsource_ex.ex [hunk 1]
```diff
@@ -8,15 +8,27 @@ defmodule EventsourceEx do
...
   end
 
-  def init(opts \\ []) do
+  defp parse_options(opts) do
     url = opts[:url]
     headers = opts[:headers]
     parent = opts[:stream_to]
+    follow_redirect = opts[:follow_redirect]
+    ssl = opts[:ssl]
+    options = [stream_to: self(), ssl: ssl, follow_redirect: follow_redirect,
```
Comment (cwc) [inline lib/eventsource_ex.ex]:
```suggestion
    http_options = [stream_to: self(), ssl: ssl, follow_redirect: follow_redirect,
```
Renaming the var helps clarify why the list is being built.

File: lib/eventsource_ex.ex [hunk 2]
```diff
@@ -8,15 +8,27 @@ defmodule EventsourceEx do
...
+                                  recv_timeout: :infinity]
+
+    {url, headers, parent, Enum.filter(options, fn({_,val}) -> val != nil end)}
+  end
+    
 
-    HTTPoison.get!(url, headers, stream_to: self(), recv_timeout: :infinity)
+
+  # todo build a list of args that exist
```
Comment (cwc) [inline lib/eventsource_ex.ex]: This comment can be deleted since you did it. 😄

**Comment by AtibaBryan** [comment]
Sorry about the belated changes I was in a meeting all day. 🤷‍♂️

**Comment by cwc** [comment]
Hey no sweat, it's rare I find time to mess with my GH projects at all these days! 😅Thanks a ton for the contribution. I'll try to push these changes to Hex soon. Given the changes to dependencies and the linking behavior, it'll be a new major version.

**Pull request merged**